test/e2e/upgrades/serviceaccount_admission_controller_migration.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package upgrades
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/onsi/ginkgo"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+)
+
+const (
+	podBeforeMigration = "pod-before-migration"
+	podAfterMigration  = "pod-after-migration"
+)
+
+// ServiceAccountAdmissionControllerMigrationTest test that a pod is functioning before and after
+// a cluster upgrade.
+type ServiceAccountAdmissionControllerMigrationTest struct {
+	pod *v1.Pod
+}
+
+// Name returns the tracking name of the test.
+func (ServiceAccountAdmissionControllerMigrationTest) Name() string {
+	return "[sig-auth] serviceaccount-admission-controller-migration"
+}
+
+// Setup creates a pod.
+func (t *ServiceAccountAdmissionControllerMigrationTest) Setup(f *framework.Framework) {
+	t.pod = t.createPod(f, podBeforeMigration)
+}
+
+// Test waits for the upgrade to complete, and then verifies that a pod.
+func (t *ServiceAccountAdmissionControllerMigrationTest) Test(f *framework.Framework, done <-chan struct{}, upgrade UpgradeType) {
+	ginkgo.By("Waiting for upgrade to finish")
+	<-done
+
+	ginkgo.By("Starting post-upgrade check")
+	pod, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Get(context.TODO(), podBeforeMigration, metav1.GetOptions{})
+	framework.ExpectNoError(err)
+	ginkgo.By(fmt.Sprintf("Got pod %q", pod.Name))
+
+	if pod.GetUID() != t.pod.GetUID() {
+		framework.Failf("Pod %q GetUID() = %q, want %q.", pod.Name, pod.GetUID(), t.pod.GetUID())
+	}
+	if pod.Status.ContainerStatuses[0].RestartCount != 0 {
+		framework.Failf("Pod %q RestartCount = %d, want 0.", pod.Name, pod.Status.ContainerStatuses[0].RestartCount)
+	}
+	if len(pod.Spec.Volumes) != 1 || pod.Spec.Volumes[0].Secret == nil {
+		framework.Failf("Pod %q Volumes[0].Secret = nil, want non-nil.", pod.Name)
+	}
```

Comment:
rather than checking for specific characteristics of the pod's service account token, I think we want to check instead that the pod's token (however it was injected) is still working properly... that means executing something in the context of the pod (like the agnhost in-cluster-client subcommand) and making sure it's not getting 401/403 errors

Reply:
done